kong/pdk/client/tls.lua

```diff
@@ -45,22 +46,42 @@ local function new()
   -- To find out whether the client honored the request, use
   -- `get_full_client_certificate_chain` in later phases.
   --
+  -- The `ca_certs` argument is the CA certificate chain opaque pointer (if any),
+  -- which can be created by the [parse_pem_cert](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/ssl.md#parse_pem_cert)
+  -- or [resty.opensslx509.chain](https://github.com/fffonion/lua-resty-openssl#restyopensslx509chain)
+  -- The list of certificates will be sent to clients. Also, they will be added
+  -- to trusted store. If omitted, will not send any CA certificate to clients.
+  --
+  -- The `depth` argument is the limit of verification depth. If omitted, the
+  -- value specified by `ssl_verify_depth` will be used. For more detail about
+  -- the `depth`, please see [SSL_set_verify_depth](https://www.openssl.org/docs/man1.1.1/man3/SSL_set_verify_depth.html#:~:text=SSL_CTX_set_verify_depth()%20and%20SSL_set_verify_depth,trust%20anchor%20certificate.)
+  --
   -- @function kong.client.tls.request_client_certificate
   -- @phases certificate
-  -- @treturn true|nil Returns `true` if request is received, or `nil` if
-  -- request fails.
-  -- @treturn nil|err Returns `nil` if the handshake is successful, or an error
-  -- message if it fails.
+  -- @tparam[opt] cdata ca_certs The CA certificate chain opaque pointer
+  -- @tparam[opt] number depth The verification depth in the client certificates chain
+  -- @treturn true|nil Returns `true` if successful, or `nil` if it fails.
+  -- @treturn nil|err Returns `nil` if successful, or an error message if it fails.
   --
   -- @usage
-  -- local res, err = kong.client.tls.request_client_certificate()
+  -- local x509_lib = require "resty.openssl.x509"
+  -- local chain_lib = require "resty.openssl.x509.chain"
+  -- local res, err
+  -- local chain = chain_lib.new()
+  -- -- err check
+  -- local x509, err = x509_lib.new(pem_cert, "PEM")
+  -- -- err check
+  -- res, err = chain:add(x509)
+  -- -- err check
+  -- -- `chain.ctx` is the raw data of the chain, i.e. `STACK_OF(X509) *`
+  -- res, err = kong.client.tls.request_client_certificate(chain.ctx, 3)
   -- if not res then
   --   -- do something with err
   -- end
-  function _TLS.request_client_certificate()
+  function _TLS.request_client_certificate(ca_certs, depth)
```

Comment:
We don't need `depth` for `mtls-auth` to work properly right? Suggest to remove it until we do have a use for it.

Also, we can not change the meaning of the return value without introducing a breaking change. Please make sure the changes to `@treturn` are intended.

Reply:
> We don't need depth for mtls-auth to work properly right? Suggest to remove it until we do have a use for it.
Yes, we can set a minimum default value.

> Also, we can not change the meaning of the return value without introducing a breaking change. Please make sure the changes to @treturn are intended.
The original description is not so accurate.  The handshake is not successful yet, only the function call is successful.

Reply:
For our current purpose, we can even set `depth` to `0` so that verification can stop early on. We don't need to actually need to verify anything by Nginx and it just wastes CPU cycles.

Reply:
Totally agree.